docs/source/API/core/view/view.rst

```diff
@@ -1,65 +1,64 @@
 ``View``
-========
+============
 
 .. role:: cppkokkos(code)
     :language: cppkokkos
 
 Header File: ``<Kokkos_Core.hpp>``
 
-Usage
------
+Description
+--------------
+
+The programming model's main data structure is the Kokkos View.  The View has multi-dimensional (up to eight) array-like semantics and operations.  It, and related data structures (*e.g.*, `View-like Types <view_like.html>`__)  can be used on the CPU (host) and/or GPU (device), and has architecture-specific optimiztions, shared-memory parallelism, compile-time layouts and memory space.   
 
-Kokkos View is a potentially reference counted multi dimensional array with compile time layouts and memory space.
-Its semantics are similar to that of ``std::shared_ptr``.
 
 Interface
----------
+------------
 
 .. code-block:: cpp
 
-    template <class DataType [, class LayoutType] [, class MemorySpace] [, class MemoryTraits]>
+    template <class DataType [, class Space] [, class Layout] [, class MemoryTraits]>
     class View;
 
 Parameters
 ~~~~~~~~~~
 
 .. _LayoutRight: layoutRight.html
 
-.. |LayoutRight| replace:: ``LayoutRight()``
+.. |LayoutRight| replace:: :cppkokkos:func:``LayoutRight``
 
 .. _LayoutLeft: layoutLeft.html
 
-.. |LayoutLeft| replace:: ``LayoutLeft()``
+.. |LayoutLeft| replace:: :cppkokkos:func:``LayoutLeft``
 
 .. _LayoutStride: layoutStride.html
 
-.. |LayoutStride| replace:: ``LayoutStride()``
+.. |LayoutStride| replace:: :cppkokkos:func:``LayoutStride``
+
+The ``DataType`` template parameter is required, but the others are optional.  Ordering is enforced for only the first (``DataType``) and last (``MemoryTraits``) positions.  That means, for example, that ``Layout`` can be omitted, but if both ``MemorySpace`` and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTraits``.  Ordering is therefore not strongly enforced for the ``Space`` or ``ArrayLayout`` parameters.
```

Comment:
This sentence is wrong. If you provide both Space and Layout, Layout has to come first.